cocos2d/shaders/CCGLProgram.js

```diff
@@ -42,28 +42,22 @@ cc.GLProgram = cc.Class.extend(/** @lends cc.GLProgram# */{
     _projectionUpdated: -1,
 
     // Uniform cache
-    _updateUniformLocation: function (location) {
-        if (!location)
+    _updateUniform: function (name) {
+        if (!name)
             return false;
 
-        var updated;
-        var element = this._hashForUniforms[location];
-
-        if (!element) {
-            element = [
-                arguments[1],
-                arguments[2],
-                arguments[3],
-                arguments[4]
-            ];
-            this._hashForUniforms[location] = element;
+        var updated = false;
+        var element = this._hashForUniforms[name];
+        var args = Array.isArray(arguments[1]) ? arguments[1] :
+            Array.prototype.slice.call(arguments, 1);
+
+        if (!element || element.length !== args.length) {
+            this._hashForUniforms[name] = args.slice();
```

Comment:
Sorry, I don't get it. Why is it necessary to copy the args array ? It's a local variable, and it's not used anymore after this line.

Reply:
```javascript
if (Array.isArray(arguments[1])) {
    args = arguments[1];
} else {
    ...
}
```
when `arguments[1]` is an array, you need to copy it, otherwise external modifications to the array will change the cache. as `(!element || element.length !== args.length)` doesn't happen often, i don't think it needs another `Array.isArray` check, just a shallow copy.

Reply:
I see, you are right, I missed this condition